src/System.Private.CoreLib/shared/System/MarshalByRefObject.cs

```diff
@@ -2,8 +2,12 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Runtime.InteropServices;
+
 namespace System
 {
+    [ClassInterface(ClassInterfaceType.AutoDispatch)]
```

Comment:
Is `ClassInterfaceAttribute` really necessary here? It was not necessary in .NET Framework...

Reply:
It isn't since that is the default in all cases, but I wanted to match `Object` since these two classes generally serve as the definitive base class in some scenarios.

Reply:
System.Object has these attributes in the non-shared partition, but MarshalByRefObject has these in the shared partition. It would be nice if the attributes were on the same side in both cases.

Reply:
@jkotas That makes sense. The `MarshalByRefObject` isn't partial, can I move the attributes for `Object` into the shared partition or do I need to make a `MarshalByRefObject` partial in the non-shared partition?

Reply:
> move the attributes for `Object` into the shared partition

Yes, I think that would make sense.